cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -453,21 +452,220 @@ private KafkaRebalanceStatus buildRebalanceStatusFromPreviousStatus(KafkaRebalan
+
+
+                    JsonObject intStats = new JsonObject();
+                    intStats.put("before", intBeforeStat);
+                    intStats.put("after", intAfterStat);
+                    intStats.put("diff", intDiff);
+
+                    brokerStats.put(intLoadParameter.getKafkaRebalanceStatusKey(), intStats);
+                } else {
+                    log.warn("{} information was missing from the broker before/after load information",
+                            intLoadParameter.getKafkaRebalanceStatusKey());
+                }
+
+            }
+
+            for (CruiseControlLoadParameters doubleLoadParameter : CruiseControlLoadParameters.getDoubleParameters()) {
+
+                if (brokerBefore.containsKey(doubleLoadParameter.getKafkaRebalanceStatusKey()) &&
+                        brokerAfter.containsKey(doubleLoadParameter.getKafkaRebalanceStatusKey())) {
+
+                    double doubleBeforeStat = (double) brokerBefore.get(doubleLoadParameter.getKafkaRebalanceStatusKey());
+                    double doubleAfterStat = (double) brokerAfter.get(doubleLoadParameter.getKafkaRebalanceStatusKey());
+                    double doubleDiff = doubleAfterStat - doubleBeforeStat;
+
+                    JsonObject doubleStats = new JsonObject();
+                    doubleStats.put("before", doubleBeforeStat);
+                    doubleStats.put("after", doubleAfterStat);
+                    doubleStats.put("diff", doubleDiff);
+
+                    brokerStats.put(doubleLoadParameter.getKafkaRebalanceStatusKey(), doubleStats);
+                } else {
+                    log.warn("{} information was missing from the broker before/after load information",
+                            doubleLoadParameter.getKafkaRebalanceStatusKey());
+                }
+
+            }
+
+            brokersStats.put(String.valueOf(loadBeforeEntry.getKey()), brokerStats);
+        }
+
+        return brokersStats;
+    }
+
+    static class MapAndStatus<T, K> {
+
+        T loadMap;
+
+        K status;
+
+        public T getLoadMap() {
+            return loadMap;
+        }
+
+        public void setLoadMap(T loadMap) {
+            this.loadMap = loadMap;
+        }
+
+        public K getStatus() {
+            return status;
+        }
+
+        public void setStatus(K status) {
+            this.status = status;
+        }
+
+        public MapAndStatus(T loadMap, K status) {
+            this.loadMap = loadMap;
+            this.status = status;
+        }
+    }
+
+    /**
+     * Converts the supplied JSONObject containing the response from the {@link CruiseControlApi#rebalance} or
+     * {@link CruiseControlApi#getUserTaskStatus} methods, into a map linking to a proposal summary map and a broker
+     * load map.
+     *
+     * @param  proposalJson The JSONObject representing the response from the Cruise Control rebalance endpoint.
+     * @return A map containing the proposal summary and broker load maps.
```

Comment:
Again, can we make sure the javadoc is accurate. This returns a `MapAndStatus` instance not a map.